pyproject.toml

```diff
@@ -79,7 +79,7 @@ dependencies = [
     # cause API breakages). If the package uses calver, we don't pin the upper
     # version, as the upper version provides no basis for determining API
     # stability.
-    "cookiecutter >= 2.3.1, < 3.0",
+    "cookiecutter >= 2.6.0, < 3.0",
```

Comment:
No objection to doing this bump if it's needed, but I'm interested what bug/feature has required it?

Reply:
Using pre-2.6.0 and 2.6.0 produces two different files for `app.py` and `__main__.py`. I didn't track down the exact thing that changed but 2.6.0 introduced a way to control the jinja tag boundaries....so, thinking about it now, maybe a regression was introduced around how `{% %}` is replaced in a template....I should probably check how our other templates look now.

At any rate, CI was failing with 2.6.0: https://github.com/beeware/briefcase/actions/runs/7993409808/job/21829525533

Reply:
We should hold off. There's all kinds on new newlines in `pyproject.toml` with 2.6.0.

Reply:
well, actually, i guess it doesn't matter....2.6.0 is going to be installed regardless...so, I guess we may as well move forward and resolve this regression independently.

Reply:
https://github.com/cookiecutter/cookiecutter/issues/2023

Reply:
To explicitly lay it out, I think these are the options I see to move forward:
- Bump to require 2.6.0 as I have here
  - This would accommodate the whitespace changes in 2.6.0
  - Once/if these are resolved in `cookiecutter`, we can revert the changes and bump to that new version
- Lower the upper bound from `<3.0` to `<2.6.0`
  - This will obviously maintain the status quo
  - Once/if the issue is resolved, we can increase the upper bound again

Let me know what you think. As a note, CI for all PRs for Briefcase will fail until this is addressed.

Reply:
I guess a third option is "!=2.6.0", if we have reason to believe that 2.6.0 is an isolated buggy release and the issue will be fixed in 2.6.1.

My inclination for now is to use "< 2.6.0" until we know whether this is an intentional change (or, at least, a change that won't be rolled back), or a bug that will be resolved. If this is the "new normal", then we should set the 2.6.0 as the new minimum; if it's a bug that gets fixed, we can exclude the specific problem releases.

Reply:
Good deal; I agree...especially since the issue is more widespread than I initially thought.
- https://github.com/beeware/briefcase/issues/1663